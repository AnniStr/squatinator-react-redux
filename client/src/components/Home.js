import React, { Component } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Container, Jumbotron, Button } from 'reactstrap';

// CSS
import './styles/Home.css'
class Home extends Component {
    render() {
        return (
            <div className="Home">
                <Jumbotron className="jumboHome">
                    <Container>
                        <h3>Welcome to Squatinator</h3>
                        <p>Now you can start creating your own and super indivdual training.</p>
                        <p>First you can create your favorite exercises and the workouts you want to do. In the second step add your exercises to the workout and set how many repetitions you wanna do in one set and how many sets you wanna do in your training. Also you can set the repetition maximum.</p>
                        <h5>Don't wait, start now!</h5>
                        <Button
                            href="/workouts"
                            className="nav-btn"
                            color="warning"
                            block>
                            Workouts
                        </Button>
                        <Button
                            href="/exercises"
                            className="nav-btn"
                            color="warning"
                            block>
                            Exercises
                        </Button>
                    </Container>
                </Jumbotron>
            </div>
        );
    }
}

export default Home;
