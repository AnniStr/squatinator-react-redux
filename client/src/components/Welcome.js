import React, { Component } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import {
    Button,
    Container,
    Jumbotron
} from 'reactstrap';

// CSS
import './styles/Welcome.css';

class Welcome extends Component {
    
    render() {
        return (
            <div className="Welcome">
            <Jumbotron fluid className="jumbo">
                <Container className="container">
                    <h2>Welcome to Squatinator!</h2>
                    <p>The fitness app to create your own individual workoutplans.</p>
                    <Button 
                        className="login-btn" 
                        color="warning" 
                        size="lg" 
                        href="/login" 
                        block>
                        Login
                    </Button>
                    <Button 
                        className="login-btn" 
                        color="warning" 
                        size="lg" 
                        href="/register" 
                        block>
                        Register now
                    </Button>
                </Container>
            </Jumbotron> 
            </div>
        );
    }
}

export default Welcome;
