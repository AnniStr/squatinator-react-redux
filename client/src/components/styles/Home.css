h3 { 
    color: black;
    text-align: center;
}

.jumboHome { 
    background: rgba(235, 233, 233, 0.7); 
    color: black;
    max-width: 800px;
    margin-left: auto; 
    margin-right: auto;
    margin-bottom: 50px;      
    border-radius: 15px;
}

.nav-btn {
    margin-left: auto; 
    margin-right: auto; 
    max-width: 200px;
}

.con {
    min-height: 250px;
    margin-top: 15px;
    margin-bottom: 50px;
}

h5 {
    text-align: center;
    margin-bottom: 20px;
}
