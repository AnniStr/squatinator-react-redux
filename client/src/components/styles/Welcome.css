.jumbo {  
    background: rgba(235, 233, 233, 0.7); 
    color: black;
    max-width: 800px;
    margin-left: auto; 
    margin-right: auto;  
    margin-bottom: 50px;      
    border-radius: 15px;
}
h2 { 
    text-align: center; 
    color: black;
}
p { 
    text-align: center; 
}
.login-btn {
    margin-left: auto; 
    margin-right: auto; 
    max-width: 400px;
}