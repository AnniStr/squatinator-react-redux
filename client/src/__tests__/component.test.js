import React from "react";
import { create } from "react-test-renderer";

import Home from "../components/Home";
import Welcome from "../components/Welcome";

describe("Home component", () => {
    test("it matches the snapshot", () => {
        const component = create(<Home/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});

describe("Welcome component", () => {
    test("it matches the snapshot", () => {
        const component = create(<Welcome/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
});