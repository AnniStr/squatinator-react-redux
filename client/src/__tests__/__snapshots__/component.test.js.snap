// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component it matches the snapshot 1`] = `
<div
  className="Home"
>
  <div
    className="jumboHome jumbotron"
  >
    <div
      className="container"
    >
      <h3>
        Welcome to Squatinator
      </h3>
      <p>
        Now you can start creating your own and super indivdual training.
      </p>
      <p>
        First you can create your favorite exercises and the workouts you want to do. In the second step add your exercises to the workout and set how many repetitions you wanna do in one set and how many sets you wanna do in your training. Also you can set the repetition maximum.
      </p>
      <h5>
        Don't wait, start now!
      </h5>
      <a
        aria-label={null}
        className="nav-btn btn btn-warning btn-block"
        href="/workouts"
        onClick={[Function]}
      >
        Workouts
      </a>
      <a
        aria-label={null}
        className="nav-btn btn btn-warning btn-block"
        href="/exercises"
        onClick={[Function]}
      >
        Exercises
      </a>
    </div>
  </div>
</div>
`;

exports[`Welcome component it matches the snapshot 1`] = `
<div
  className="Welcome"
>
  <div
    className="jumbo jumbotron jumbotron-fluid"
  >
    <div
      className="container container"
    >
      <h2>
        Welcome to Squatinator!
      </h2>
      <p>
        The fitness app to create your own individual workoutplans.
      </p>
      <a
        aria-label={null}
        className="login-btn btn btn-warning btn-lg btn-block"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
      <a
        aria-label={null}
        className="login-btn btn btn-warning btn-lg btn-block"
        href="/register"
        onClick={[Function]}
      >
        Register now
      </a>
    </div>
  </div>
</div>
`;
